examples/advanced/backup.yaml

```diff
@@ -0,0 +1,260 @@
+apiVersion: pingcap.com/v1alpha1
+kind: Backup
+metadata:
+  name: advanced-backup
+  namespace: default
+
+spec:
+  ## Describes the compute resource requirements and limits of Backup.  
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/
+  requests:
+    cpu: 1000m
+    memory: 1Gi
+  limits:
+    cpu: 2000m
+    memory: 2Gi
+  
+  ## List of environment variables to set in the container, like v1.Container.Env.
+  ## Note that the following builtin env vars will be overwritten by values set here.
+  ## - S3_PROVIDER
+  ## - S3_ENDPOINT
+  ## - AWS_REGION
+  ## - AWS_ACL
+  ## - AWS_STORAGE_CLASS
+  ## - AWS_DEFAULT_REGION
+  ## - AWS_ACCESS_KEY_ID
+  ## - AWS_SECRET_ACCESS_KEY
+  ## - GCS_PROJECT_ID
+  ## - GCS_OBJECT_ACL
+  ## - GCS_BUCKET_ACL
+  ## - GCS_LOCATION
+  ## - GCS_STORAGE_CLASS
+  ## - GCS_SERVICE_ACCOUNT_JSON_KEY
+  ## - BR_LOG_TO_TERM
+  # env: []
+
+  ## From is the TidbCluster to be backed up.
+  from:
+    ## Host is the address of the TidbCluster to be backed up, which is the service name of the TiDB cluster to be exported, such as `basic-tidb`.
+    host: advanced-tidb-tidb
+    ## Port is the port of the TidbCluster to be backed up.
+    port: 4000
+    ## User is the accessing user of the TidbCluster to be backed up.
+    user: backup
+    ## SecretName is the secret that contains the password of the accessing user of the TidbCluster to be backed up.
+    secretName: backup-secret
+    ## TLSClientSecretName is the name of secret which stores tidb server client certificate.
+    ## Defaults to nil.
+    # tlsClientSecretName: ""
+  
+  ## BackupType is the backup type for TidbCluster.
+  ## `full` represents the full backup of TidbCluster.
+  ## `raw` represents the raw backup of TidbCluster.
+  ## `db` represents the backup of one DB of TidbCluster.
+  ## `table` represents the backup of one table of TidbCluster.
+  ## `tiflash-replica` represents restoring the tiflash replica removed by a failed restore of the older version BR.
```

Comment:
this seems not valid now, ref https://github.com/pingcap/br/blob/c7f93829ef433dea4a86d0eb506a19a1339e4201/cmd/br/backup.go#L89-L92

https://github.com/pingcap/br/blob/c7f93829ef433dea4a86d0eb506a19a1339e4201/cmd/br/restore.go#L103-L107

Reply:
`backupType` is used in the controller now,  shall we modify this when we remove them from the controller?

Reply:
I mean, `tiflash-replica` is not valid in BR

Reply:
Get it, fixed.